lib/utils/location.js

```diff
@@ -75,18 +75,27 @@ const appPageTypes: { [AppType]: {| default?: PageType, pageTypes: PageType[] |}
 		pageTypes: ['wiki', 'search', 'stylesheet', 'modqueue', 'subredditAbout', 'comments', 'commentsLinklist', 'profile', 'liveThread', 'inbox', 'submit', 'account', 'prefs'],
 	},
 	d2x: {
-		pageTypes: ['profile2x', 'profileCommentsPage'],
+		pageTypes: ['profile2x', 'profileCommentsPage', 'submit'],
 	},
 	options: {
 		pageTypes: [],
 	},
 };
 
-export const pageType = _.once((): ?PageType => {
+export const pageType = _.memoize((): ?PageType => {
+	// Using waitForEvent from utils/dom.js would throw an error
+	// during the test for this module as $ is defined on
+	// the window in that module. Emulating functionality here
```

Comment:
I didn't realize that. I have removed the $ dependency f78d299d5840281dc92a5c8f48502166952f5c80, so I think `waitForEvent` can be used from `location` now without breaking the test.